### PR TITLE
Handle zaps

### PR DESCRIPTION
**Pull request by blastshielddown**

### CODE REVIEW

File: db/migrations/20240202223238_add_is_pending_to_ext_receive.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+exports.up = function (knex) {
+  return knex.schema.alterTable("external_receive", function (table) {
+    table.boolean("is_pending").defaultTo(false); // Set default to false for existing records
```
Comment (joshr4) [inline db/migrations/20240202223238_add_is_pending_to_ext_receive.js:3]: for the `external_payment` migration, I defaulted this to true, thought process was that every record would start off as pending and then later need to be switched to false. Doesn't seem super significant though as long as we set the value and not rely on the default?

File: library/content.js [hunk 2]
```diff
@@ -85,3 +85,17 @@ export const getContentFromId = async (contentId) => {
 
   return content;
 };
+
+export const getContentFromEventId = async (eventId) => {
+  const { content_id } = await db
+    // the event_track table used to be only for tracks, but now it's for all content types
+    .knex("event_track")
+    // track_id is really content_id (tracks/episodes/podcasts/al)
```
Comment (joshr4) [inline library/content.js:93]:
```suggestion
    // track_id is really content_id (tracks/episodes/podcasts/album)
```

File: library/invoice.ts [hunk 3]
```diff
@@ -33,10 +35,75 @@ export const updateInvoiceIfNeeded = async (
...
+  const isZap = false;
+  const isPartyMode = false;
+
+  // const paymentTypes = {
+  //   6: "invoice",
+  //   7: "zap",
+  //   8: "party",
+  // };
+
+  const paymentType = isZap ? 7 : isPartyMode ? 8 : 6;
```
Comment (joshr4) [inline library/invoice.ts:72]: these type numbers might be good to type as an enum
```
enum PaymentType {
  Zap = 7,
  PartyMode = 8,
  Invoice = 6,
}
```

File: prisma/schema.prisma [hunk 4]
```diff
@@ -481,7 +482,7 @@ model ExternalPayment {
   createdAt    DateTime? @default(now()) @map("created_at") @db.Timestamptz(6)
   updatedAt    DateTime? @default(now()) @map("updated_at") @db.Timestamptz(6)
   external_id  Boolean?
-  isPending    Boolean?  @default(true) @map("is_pending")
+  is_pending   Boolean?  @default(true)
```
Comment (joshr4) [inline prisma/schema.prisma:485]: looks like the `isPending` alias was undone here?

File: services/accounting/package.json [hunk 5]
```diff
@@ -20,7 +20,7 @@
     "dotenv": "^16.1.3",
     "express": "^4.18.2",
     "loglevel": "^1.7.1",
-    "nostr-tools": "^1.17.0",
+    "nostr-tools": "^1.8.2",
```
Comment (joshr4) [inline services/accounting/package.json:23]: is this a downgrade? looks like the latest version is `2.1.5`, we may want to upgrade

File: db/migrations/20240202223238_add_is_pending_to_ext_receive.js (hunk 1 above)
Comment (blastshielddown) [reply db/migrations/20240202223238_add_is_pending_to_ext_receive.js:3]: yes, so i had originally defaulted to `true` here too just like in `external_payment`, but then realized that would incorrectly label all existing and settled payments in the table as `is_pending` (when, in fact, every single one of those records was settled). so for the sake of those records we're defaulting to `false` and then making sure to set every new record to `true` at create time

File: services/accounting/package.json (hunk 5 above)
Comment (blastshielddown) [reply services/accounting/package.json:23]: oh you're right, i'll switch back to latest `1.x` release. I believe `2.x` has breaking changes so will stick to 1 for now

**Review by joshr4 — APPROVED** [review]

**Pull request merged**